Sources/Data Model/ProjectConfig.swift

```diff
@@ -122,6 +124,44 @@ class ProjectConfig {
             project.rollouts.forEach { map[$0.id] = $0 }
             return map
         }()
+        
+        // all rules (experiment rules and delivery rules) for each flag
+        
+        self.flagRulesMap = {
+            var map = [String: [Experiment]]()
+            
+            project.featureFlags.forEach { flag in
+                var experiments = flag.experimentIds.compactMap { experimentIdMap[$0] }
+                let rollout = self.rolloutIdMap[flag.rolloutId]
+                experiments.append(contentsOf: rollout?.experiments ?? [])
+                
+                map[flag.key] = experiments
+            }
+            
+            return map
+        }()
+        
+        // all variations for each flag
+        // - datafile does not contain a separate entity for this.
+        // - we collect variations used in each rule (experiment rules and delivery rules)
+        
+        self.flagVariationsMap = {
+            var map = [String: [Variation]]()
+            
+            flagRulesMap.forEach { (flagKey, rules) in
```

Comment:
Can we include this in the initial iteration through the flags when we build the flag rules map? To avoid looping though the same experiments/rollouts twice?

Reply:
A good point. Refactored.